modules/core/include/opencv2/core/ocl.hpp

```diff
@@ -261,6 +261,8 @@ class CV_EXPORTS Context
     void setUseSVM(bool enabled);
 
     struct Impl;
+    inline Impl* getImpl() const { return (Impl*)p; }
+//protected:
```

Comment:
why `Impl* p` is not protected anymore if we have getImpl()?

Reply:
Because "p" is not protected initially. It is "TODO" mark for future releases.
Accurate testing of libva / directx / opengl integrations is required before apply this.